gulpfile.js

```diff
@@ -22,6 +23,115 @@ const rollup = require('rollup');
 const rollupString = require('rollup-plugin-string');
 const rollupTypescript = require('rollup-plugin-typescript2');
 
+const rxjsTypes = new Set(['Observer']);
+
+const transformRxjsImportText = (importName) =>
+  importName === 'EMPTY'
+    ? 'rxjs/internal/observable/empty'
+    : rxjsTypes.has(importName)
+    ? 'rxjs/internal/types'
+    : importName[0].toLowerCase() === importName[0]
+    ? `rxjs/internal/observable/${importName}`
+    : `rxjs/internal/${importName}`;
+
+const transformRxjsOperatorsImportText = (importName) => `rxjs/internal/operators/${importName}`;
+
+const transformImport = (name, callback, statements) => {
+  const idx = statements.findIndex(
+    (statement) => typescript.isImportDeclaration(statement) && statement.moduleSpecifier.text === name,
+  );
+  if (idx !== -1) {
+    const beforeIdx = statements.slice(0, idx);
+    const foundImport = statements[idx];
+    const afterIdx = statements.slice(idx + 1);
+
+    const newImports = foundImport.importClause.namedBindings.elements.map((elem) =>
+      typescript.createImportDeclaration(
+        foundImport.decorators,
+        foundImport.modifiers,
+        typescript.createImportClause(
+          undefined,
+          typescript.createNamedImports([typescript.createImportSpecifier(elem.propertyName, elem.name)]),
+        ),
+        typescript.createStringLiteral(
+          callback(elem.propertyName === undefined ? elem.name.getText() : elem.propertyName.getText()),
+        ),
+      ),
+    );
+
+    return beforeIdx.concat(newImports, afterIdx);
+  }
+
+  return statements;
+};
+
+const createCustomTransform = (context) => {
+  return (node) => {
+    const visitor = (node) => {
+      if (typescript.isSourceFile(node)) {
+        let statements = node.statements;
+        statements = transformImport(`rxjs`, transformRxjsImportText, statements);
+        statements = transformImport(`rxjs/operators`, transformRxjsOperatorsImportText, statements);
+
+        return typescript.updateSourceFileNode(
+          node,
+          statements,
+          node.isDeclarationFile,
+          node.referencedFiles,
+          node.typeReferences,
+          node.hasNoDefaultLib,
+          node.libReferences,
+        );
+      }
+
+      return node;
+    };
+
+    typescript.visitEachChild(node, visitor, context);
+
+    return typescript.visitNode(node, visitor);
+  };
+};
+
+const createCustomBrowserifyTransform = (context) => {
+  return (node) => {
+    const visitor = (node) => {
+      if (typescript.isImportDeclaration(node)) {
+        const moduleName = node.moduleSpecifier.text;
+        if (isBuiltinModule(moduleName)) {
+          return typescript.updateImportDeclaration(
+            node,
+            node.decorators,
+            node.modifiers,
+            node.importClause,
+            typescript.createStringLiteral(`${moduleName}-browserify`),
```

Comment:
You need to check that all of these modules actually exist. (Hint: they don't, so we should only be transforming the ones that do)